client/post-editor/media-modal/image-editor/image-editor-canvas.jsx

```diff
@@ -67,78 +82,136 @@ const MediaModalImageEditorCanvas = React.createClass( {
 		}
 
 		this.drawImage();
+		this.updateCanvasPosition();
 	},
 
 	componentDidUpdate() {
 		this.drawImage();
+		this.updateCanvasPosition();
 	},
 
 	toBlob( callback ) {
 		const canvas = ReactDom.findDOMNode( this.refs.canvas );
-
-		MediaUtils.canvasToBlob( canvas, callback, this.props.mimeType, 1 );
+		const context = canvas.getContext( '2d' );
+		const rotated = this.props.transform.degrees % 180 !== 0;
+		const imageWidth = rotated ? this.image.height : this.image.width;
+		const imageHeight = rotated ? this.image.width : this.image.height;
+		const croppedLeft = this.props.crop.leftRatio * imageWidth;
+		const croppedTop = this.props.crop.topRatio * imageHeight;
+		const croppedWidth = this.props.crop.widthRatio * imageWidth;
+		const croppedHeight = this.props.crop.heightRatio * imageHeight;
+		const imageData = context.getImageData(
+			croppedLeft,
+			croppedTop,
+			croppedWidth,
+			croppedHeight );
+		const newCanvas = document.createElement( 'canvas' );
+
+		newCanvas.width = croppedWidth;
+		newCanvas.height = croppedHeight;
+
+		const newContext = newCanvas.getContext( '2d' );
+		newContext.putImageData( imageData, 0, 0 );
+
+		MediaUtils.canvasToBlob( newCanvas, callback, this.props.mimeType, 1 );
 	},
 
 	drawImage() {
 		if ( ! this.image ) {
 			return;
 		}
 
-		if ( this.state.canvasWidth !== this.image.width ||
-			this.state.canvasHeight !== this.image.height ) {
-			this.setState( {
-				canvasWidth: this.image.width,
-				canvasHeight: this.image.height
-			} );
-		}
+		const canvas = ReactDom.findDOMNode( this.refs.canvas );
+		const imageWidth = this.image.width;
+		const imageHeight = this.image.height;
+		const transform = this.props.transform;
+		const rotated = transform.degrees % 180 !== 0;
+
+		//make sure the canvas draw area is the same size as the image
+		canvas.width = rotated ? imageHeight : imageWidth;
```

Comment:
Yes, my plan is to move this as well as the aspect ratio controlling logic in image-editor-crop.js to a reducer or at least some common util function. It will be a part of the improvements that include animations, since they might make a lot of this logic redundant/more centralized.